webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala

```diff
@@ -1281,8 +1360,31 @@ object JsonResponseObjects {
         renderInHtml:          RenderInheritedProperties,
         escapeHtml:            Boolean = false
     ): JRProperty = {
-      val (parents, origval) = prop.hierarchy.reverse match {
-        case Nil  => (None, None)
+      val hierarchyStatus = JRPropertyHierarchyStatus.fromParentProperties(hasChildTypeConflicts, fullHierarchy)
+      val desc            = if (prop.prop.description.trim.isEmpty) None else Some(prop.prop.description)
+      JRProperty(
+        prop.prop.name,
+        prop.prop.value,
+        desc,
+        prop.prop.inheritMode,
+        prop.prop.provider,
+        transformParentProperties(prop.hierarchy, renderInHtml, escapeHtml),
+        hierarchyStatus,
+        getHierarchyOriginalValue(prop.hierarchy)
+      )
+    }
+
+    /**
+     * Transform proprety hierarchy into HTML or JSON
+     */
+    private def transformParentProperties(
+        hierarchy:    List[ParentProperty],
+        renderInHtml: RenderInheritedProperties,
+        escapeHtml:   Boolean
+    ): Option[JRPropertyHierarchy] = {
+      // FIXME: instead of having a reverse hierarchy we should have the list sorted, and reflected in the type
```

Comment:
FIXME should never reach a PR appart to find a way to remove them :)
See next comment for that one.